### ISSUE TITLE
Why can't we make tcpdump to not stop/exit when an interface is brought down?

### ISSUE THREAD

**Issue by vaibhavreddys** [the-tcpdump-group/libpcap#1214]
tcpdump if started with _-i <specific_interace_name>_ exits if that <specific interface> is brought down.
However, if it is started with "_-i any_" option, it doesn't stop if _wlanX/ethX_ interfaces are removed because the other interfaces (like lo, dummy, rmnet) are present.

And on my Android device if I want to capture packets on my wlan0 interfaces for a longer period(which includes multiple WiFi restarts in between), with the current implementation, I wouldn't be able to do it because tcpdump stops when the WiFi is turned OFF.

So how about we implement something similar to "_-i any_". Many linux machines(at least _arm64_ based android devices) have _dummy0_ interface, so with a special option, we provide a functionality where we can capture packets on _wlan0_ interaface and _dummy0_ interface (although _dummy0_ interface doesn't produce any packets), so that when _wlan0_ interface is brought down as part of the WiFi OFF operation we will still have _dummy0_ interface and tcpdump doesn't exit.

(Not sure, if I have to make this a feature request or an issue)

**Comment by guyharris** [comment]
As @infrastation notes, there have been changes to libpcap to distringuish between "this interface was brought down but is still present" and "this interface was removed".

There is no reason to continue capturing on an interface that no longer exists.

**Comment by guyharris** [comment]
I tested this on Ubuntu 20.04, which has libpcap 1.9.1 and thus doesn't have the change @infrastation mentioned, and on Ubuntu 22.04, which has libpcap 1.10.1 and thus *does* have that change, by running a tcpdump in one window and, in another window, using `ifconfig` to configure the interface on which tcpdump was capturing down.

The capture got an error on Ubuntu 20.04; it did *not* get an error on Ubuntu 22.04 and, if I configured the interface back up again, capturing resumed.

So it's fixed.
